git/git.go

```diff
@@ -87,6 +87,16 @@ func RemoteShaChanged(url, branch, sha, uuid string) bool {
 	return true
 }
 
+func LatestPathCommit(path string) (string, error) {
+	pathSegments := strings.Split(path, "/")
+	gitRoot := fmt.Sprintf("/var/lib/cattle/%s", strings.Join(pathSegments[0:3], "/"))
+
+	cmd := exec.Command("git", fmt.Sprintf("--git-dir=%s/.git", gitRoot), "-C", gitRoot, "log", "-1", "--oneline", "--format=format:%H", "--", fmt.Sprintf("/var/lib/cattle/%s", path))
```

Comment:
This git command needs to run in the respective git directory to get the history for changes in that directory.